core/extensions/impl/memory_extension.cpp

```diff
@@ -6,13 +6,34 @@
 #include <exception>
 
 #include "extensions/impl/memory_extension.hpp"
+#include "memory_extension.hpp"
 
 namespace kagome::extensions {
-  uint8_t *MemoryExtension::ext_malloc(uint32_t size) {
-    std::terminate();
+  MemoryExtension::MemoryExtension()
+      : logger_{common::createLogger("WASM Runtime [MemoryExtension]")} {}
```

Comment:
we need to get rid of this habit of initializing loggers in-place. 

logger is also a dependency and we should be able to instantiate it with custom logger.

you can define 2 constructors to simplify
```
MemoryExtension(): this(common::createLogger(...)) {...}
MemoryExtension(const Logger &logger): logger_(logger){}
```

Reply:
I vote for default arg logger